### ISSUE TITLE
Credential must have exactly 5 slash-delimited elements

### ISSUE THREAD

**Issue by mbonig** [chiefbiiko/dynamodb#13]
Simple test:

```
import { createClient } from "https://raw.githubusercontent.com/mbonig/dynamodb/master/mod.ts";

const ddb = createClient({port:443});

const results = await ddb.listTables();
```
Results in this error:

```

error: Uncaught Error: Credential must have exactly 5 slash-delimited elements, e.g. keyid/date/region/service/term, got '/'
```

Note the above code uses a slight modification of this module due to the other issues I've mentioned previously.

**Comment by mbonig** [comment]
Ok, well I can tell you how I fixed this, but I don't think it's right:

```
export function createClient(conf?: ClientConfig): DynamoDBClient {
  const _conf: Doc = deriveConfig(conf);
  // added this:
  _conf.cache.refresh();

  const dyno: DynamoDBClient = {} as DynamoDBClient;

  for (const op of OPS) {
    dyno[camelCase(op)] = baseOp.bind(null, _conf, op);
  }

  return dyno;
}

```

I say this isn't right because 
A) it doesn't await the refresh, so could be a ticking timebomb
B) I'm not convinced this is the right place to do the initial refresh (or maybe the refresh should happen immediately).

**Comment by chiefbiiko** [comment]
the `v0.3.0` release fixes those config and auth errors

**Comment by mbonig** [comment]
Issue is resolved with v0.3.0
